sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -63,41 +65,65 @@ public class AzurePipelinesCredential implements TokenCredential {
      * @param identityClientOptions the options for configuring the identity client
      */
     AzurePipelinesCredential(String clientId, String tenantId, String requestUrl, String systemAccessToken, IdentityClientOptions identityClientOptions) {
-        clientAssertionCredentialHelper = new ClientAssertionCredentialHelper(clientId, tenantId, identityClientOptions, () -> {
-            HttpClient client = identityClientOptions.getHttpClient();
-            if (client == null) {
-                HttpClient.createDefault();
-            }
-            HttpPipeline pipeline = IdentityClientBase.setupPipeline(client, identityClientOptions);
-            try {
-                URL url = new URL(requestUrl);
-                HttpRequest request = new HttpRequest(HttpMethod.POST, url);
-                request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
-                request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
-                try (HttpResponse response = pipeline.sendSync(request, Context.NONE)) {
-                    String responseBody = response.getBodyAsBinaryData().toString();
-                    if (response.getStatusCode() != 200) {
-                        throw LOGGER.logExceptionAsError(new ClientAuthenticationException("Failed to get the client assertion token "
-                            + responseBody
-                            + System.lineSeparator()
-                            + "For troubleshooting information see https://aka.ms/azsdk/java/identity/azurepipelinescredential/troubleshoot.", response));
-                    }
-                    try (JsonReader reader = JsonProviders.createReader(responseBody)) {
-                        return OidcTokenResponse.fromJson(reader).getOidcToken();
+
+        IdentityClientBuilder builder = new IdentityClientBuilder()
+            .tenantId(tenantId)
+            .clientId(clientId)
+            .identityClientOptions(identityClientOptions)
+            .clientAssertionSupplierWithHttpPipeline((httpPipeline) -> {
+                try {
```

Comment:
nit: This function can be refactored to a method in IdentityUtil to improve readability of landing public class.

Reply:
I think you mean the callback, yeah? I don't think it belongs somewhere else; this type defining it seems very reasonable to me. I could be convinced to put it in a method on this type and pass that method reference in if you felt strongly about this much code in the ctor.